support/types.go

```diff
@@ -94,12 +114,48 @@ type E2EClusterProviderWithImageLoader interface {
 	// LoadImage is used to load a set of Docker images to the cluster via the cluster provider native workflow
 	// Not every provider will have a mechanism like this/need to do this. So, providers that do not have this support
 	// can just provide a no-op implementation to be compliant with the interface
-	LoadImage(ctx context.Context, image string) error
+	LoadImage(ctx context.Context, image string, args ...string) error
 
 	// LoadImageArchive is used to provide a mechanism where a tar.gz archive containing the docker images used
 	// by the services running on the cluster can be imported and loaded into the cluster prior to the execution of
 	// test if required.
 	// Not every provider will have a mechanism like this/need to do this. So, providers that do not have this support
 	// can just provide a no-op implementation to be compliant with the interface
-	LoadImageArchive(ctx context.Context, archivePath string) error
+	LoadImageArchive(ctx context.Context, archivePath string, args ...string) error
+}
+
+// E2EClusterProviderWithImageLoaderAndNodeLifecycle is an interface that extends the E2EClusterProviderWithImageLoader
+// interface to provide a mechanism to add/remove nodes from the cluster as part of the E2E Test workflow.
+//
+// This can be useful while performing the e2e test that revolves around the node lifecycle events.
+// eg: You have a kubernetes controller that acts upon the v1.Node resource of the k8s and you want to
+// test out how the Remove operation impacts your workflow.
+// Or you want to simulate a case where one or more node of your cluster is down and you want to see how
+// your application reacts to such failure events.
+type E2EClusterProviderWithImageLoaderAndNodeLifecycle interface {
```

Comment:
Oof, that is a long name.
Let's reconsider making it simpler or break into multiple (embedded) interfaces.
* `E2EClusterProvier`
* `E2EClusterProviderWithImgLoader`
*`E2EClusterProviderWithLifeCycle`

Reply:
Ack. using `E2EClusterProviderWithLifeCycle` instead

Reply:
@vladimirvivien  Do you want to check the image loader related interface name too ?